src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -1896,7 +1896,9 @@ public SolutionState WithFrozenSourceGeneratedDocument(SourceGeneratedDocumentId
                     documentIdentity,
                     sourceText,
                     projectState.ParseOptions!,
-                    projectState.LanguageServices);
+                    projectState.LanguageServices,
+                    // Just compute the checksum from the source text passed in.
+                    originalSourceTextChecksum: null);
```

Comment:
@jasonmalinowski  i think this is correct since this is frozen partial... 

note: i don't expect we would ever need the checksum in tha tcase.  but if we do, we should use hte sourceText we froze to.

Reply:
Especially in this path: this means the user had an open source generated file, and while we're in the process of freezing the generator isn't generating that document anymore. In this case, the checksum is reasonably whatever the client has, which would get synced back over to the OOP if it's needed there.

(frankly even if there's a problem here, changing how checksums work per our call would make this not be a problem anymore and so it's probably better to spend the time on the more robust change than try to work through this edge case, especially since it's a preview)